R/utils_defaults_handling.R

```diff
@@ -0,0 +1,351 @@
+# Helper function so to modify only in one place for checks
+#' @keywords internal
+assert_allowed_types <- function(type) {
+  default_type <- c("counts", "numeric")
+  checkmate::assert_character(type, null.ok = TRUE)
+  checkmate::assert_subset(type,
+    choices = default_type,
+    empty.ok = TRUE
+  )
+  invisible(default_type)
+}
+
+#' Defaults for stats methods names and their relative formats/labels
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' Utility functions to get valid statistic methods for different method groups
+#' (`.stats`) and their relative formats (`.formats`) and labels (`.labels`). This utility
+#' is used across `tern`, but some of its working principles can be seen in [analyze_vars()].
+#' See notes to understand why this is experimental.
+#'
+#' @param stats (`character`) \cr statistical methods to get defaults formats for.
+#' @param type (`character`) \cr type of data and result desired. Some stats defaults (and their formats and labels),
```

Comment:
We usually don't put a space here, but I'm not sure if it actually makes a difference. I think you've added the space in all your new functions so you might want to change this in those as well.
```suggestion
#' @param stats (`character`)\cr statistical methods to get defaults formats for.
#' @param type (`character`)\cr type of data and result desired. Some stats defaults (and their formats and labels),
```